pinc/access_log.inc

```diff
@@ -38,3 +38,23 @@ function get_first_granted_date($username, $stage)
 
     return $row["timestamp"];
 }
+
```

Comment:
[Commenting here because github won't let me comment directly on the commit message.]

> Make an access_log record when a user privilege is granted or revoked.

That already happens. You need to add something like "via manage_site_access_privileges.php".

> This slightly abuses the access_log record, which in the past has only taken a stage name for an activity.

It's not an abuse, the access_log table was designed from the start to handle more than just stage-access. That's why the column is labelled 'activity', not 'stage'. Moreover, this isn't the first non-stage use: the table has entries with activity='DU' and activity='P2_mentor', neither of which is a stage id.

> Now it also takes other access privileges that are more verbose.

(Change "takes" to "records", I think.) So that's why you need to widen the 'activity' column. I think it would make sense to squash the two commits.

Reply:
I'll squash this and the database update together and update the commit message before merging.